openupgrade_scripts/scripts/hr_holidays/17.0.1.6/upgrade_analysis_work.txt

```diff
@@ -0,0 +1,186 @@
+hr_holidays  / hr.leave                 / date_to (datetime)            : now a function
+hr_holidays  / hr.leave                 / employee_ids (many2many)      : now a function
+# NOTHING TO DO: computed fields
+
+hr_holidays  / hr.leave                 / holiday_allocation_id (many2one): DEL relation: hr.leave.allocation
+# NOTHING TO DO: removed field
+
+hr_holidays  / hr.leave                 / message_has_sms_error (boolean): previously in module sms
+# NOTHING TO DO: Handled by Odoo registry
+
+hr_holidays  / hr.leave                 / mode_company_id (many2one)    : not a function anymore
+# NOTHING TO DO: states removed only
+
+hr_holidays  / hr.leave                 / number_of_days (float)        : now a function
+hr_holidays  / hr.leave                 / number_of_hours (float)       : NEW isfunction: function, stored
+# NOTHING TO DO: computed fields
+
+hr_holidays  / hr.leave                 / rating_ids (one2many)         : NEW relation: rating.rating
+hr_holidays  / hr.leave                 / resource_calendar_id (many2one): NEW relation: resource.calendar, hasdefault: compute
+# NOTHING TO DO: new field
+
+hr_holidays  / hr.leave                 / website_message_ids (one2many): previously in module portal
+# NOTHING TO DO: Handled by Odoo registry
+
+hr_holidays  / hr.leave.accrual.level   / action_with_unused_accruals (selection): selection_keys is now '['all', 'lost', 'maximum']' ('['lost', 'postponed']')
+# DONE: pre-migration reallocate postponed --> all or maximum depending on maximum_leave
+
+hr_holidays  / hr.leave.accrual.level   / added_value_type (selection)  : selection_keys is now '['day', 'hour']' ('['days', 'hours']')
+# DONE: pre-migration reallocate values
+
+hr_holidays  / hr.leave.accrual.level   / cap_accrued_time (boolean)    : NEW hasdefault: default
+# NOTHING TO DO: new functionality
+
+hr_holidays  / hr.leave.accrual.level   / frequency (selection)         : selection_keys is now '['bimonthly', 'biyearly', 'daily', 'hourly', 'monthly', 'weekly', 'yearly']' ('['bimonthly', 'biyearly', 'daily', 'monthly', 'weekly', 'yearly']')
+# NOTHING TO DO: new key 'hourly'
+
+hr_holidays  / hr.leave.accrual.level   / is_based_on_worked_time (boolean): DEL
+hr_holidays  / hr.leave.accrual.plan    / is_based_on_worked_time (boolean): NEW hasdefault: compute
+# DONE: pre-migration propagate value from hr.leave.accrual.level to hr.leave.accrual.plan
+
+hr_holidays  / hr.leave.accrual.level   / parent_id (many2one)          : DEL relation: hr.leave.accrual.level
+# NOTHING TO DO: removed field without replacement
+
+hr_holidays  / hr.leave.accrual.plan    / accrued_gain_time (selection) : NEW required, selection_keys: ['end', 'start'], hasdefault: default
+# NOTHING TO DO: new field with default value = v16 behaviour
+
+hr_holidays  / hr.leave.accrual.plan    / active (boolean)              : NEW hasdefault: default
+# NOTHING TO DO: new field, default = active
```

Comment:
please, do like `_pre_create_account_report_active` in pre-migration of `account`.